Rate limit changes (fixes #5287)

Nutomic opened the pull request:
Using the same rate limit as create post because it belongs together and is less verbose this way (no need for a separate service). Tested and still works without the post guard.

dessalines commented on src/api_routes_v3.rs:
```diff
@@ -199,10 +199,11 @@ pub fn config(cfg: &mut ServiceConfig, rate_limit: &RateLimitCell) {
         .service(
           // Handle POST to /post separately to add the post() rate limitter
-          resource("/post")
+          scope("/post")
```
CI failing. Maybe the fact that there are two `service(scope("/post))` is an issue

dessalines replied:
From what I can tell, this still needs to be resource. Check out the `image` block at the bottom of v4.

Nutomic commented:
The post rate limit is actually too strict with 6 per 10 minutes, so if you make 3 link posts it would already be used up. So I switched to search rate limit.

Also changed `resolve_object` to use search rate limit as message limit is too high. And in v4 the limit was missing entirely.
